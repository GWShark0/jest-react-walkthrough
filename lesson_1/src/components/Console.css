.console {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin: auto;
  transition: background-color 200ms ease-in-out;
}

.console--green {
  background-color: #4CAF50;
}

.console--yellow {
  background-color: #FFEB3B;
}

.console--red {
  background-color: #F44336;
}

.console .button:not(:last-child) {
  margin-bottom: 1rem;
}

.green-button,
.yellow-button,
.red-button {
  width: 20rem;
}

.green-button {
  background-color: #4CAF50;
  border-color: #28612B;
  color: #28612B;
}

.green-button:hover {
  background-color: #68d86e;
}

.yellow-button {
  background-color: #FFEB3B;
  border-color: #FFA61F;
  color: #FFA61F;
}

.yellow-button:hover {
  background-color: #fffa51;
}

.red-button {
  background-color: #F44336;
  border-color: #BD241D;
  color: #BD241D;
}

.red-button:hover {
  background-color: #fd5c4a;
}
