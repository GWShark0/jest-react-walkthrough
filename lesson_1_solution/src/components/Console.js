import React, { Component } from 'react';
import classNames from 'classnames';
import Button from './Button';

import './Console.css';

class Console extends Component {
  state = {
    code: ''
  }

  switchCode = (code) => {
    if (code === this.state.code) {
      this.setState({ code: '' });
    } else {
      this.setState({ code });
    }
  }

  render() {
    const className = classNames(
      'console',
      { [`console--${this.state.code}`]: !!this.state.code },
    );

    return (
      <div className={className}>
        <Button
          className="green-button"
          onClick={() => this.switchCode('green')}
        >
          Code Green
        </Button>
        <Button
          className="yellow-button"
          onClick={() => this.switchCode('yellow')}
        >
          Code Yellow
        </Button>
        <Button
          className="red-button"
          onClick={() => this.switchCode('red')}
        >
          Code Red
        </Button>
      </div>
    );
  }
}

export default Console;
