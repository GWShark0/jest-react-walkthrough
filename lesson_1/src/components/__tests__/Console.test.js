import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Console from '../Console';

describe('Console', () => {

  it('should render', () => {
    const component = renderer.create(<Console />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe('switchCode()', () => {
    it('should switch on a new code', () => {

    });

    it('should toggle off an already present code', () => {

    });

    it('should be called', () => {

    });
  });

  it('should display a code class', () => {

  });

});
