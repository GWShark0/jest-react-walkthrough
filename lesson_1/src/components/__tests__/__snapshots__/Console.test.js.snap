// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Console should render 1`] = `
<div
  className="console"
>
  <button
    className="green-button button"
    onClick={[Function]}
  >
    Code Green
  </button>
  <button
    className="yellow-button button"
    onClick={[Function]}
  >
    Code Yellow
  </button>
  <button
    className="red-button button"
    onClick={[Function]}
  >
    Code Red
  </button>
</div>
`;
